packages/dataviews/src/view-list.js

```diff
@@ -1,9 +1,96 @@
 /**
- * Internal dependencies
+ * External dependencies
  */
-import ViewTable from './view-table';
+import classNames from 'classnames';
 
-export default function ViewList( props ) {
-	// To do: change to email-like preview list.
-	return <ViewTable { ...props } />;
+/**
+ * WordPress dependencies
+ */
+import { useAsyncList } from '@wordpress/compose';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { ENTER } from '@wordpress/keycodes';
+
+export default function ViewList( {
+	view,
+	fields,
+	data,
+	getItemId,
+	onSelectionChange,
+	selection,
+} ) {
+	const shownData = useAsyncList( data, { step: 3 } );
+	const mediaField = fields.find(
+		( field ) => field.id === view.layout.mediaField
+	);
+	const primaryField = fields.find(
+		( field ) => field.id === view.layout.primaryField
+	);
+	const visibleFields = fields.filter(
+		( field ) =>
+			! view.hiddenFields.includes( field.id ) &&
+			! [ view.layout.primaryField, view.layout.mediaField ].includes(
+				field.id
+			)
+	);
+
+	const onEnter = ( item ) => ( event ) => {
+		const { keyCode } = event;
+		if ( keyCode === ENTER ) {
+			onSelectionChange( [ item ] );
+		}
+	};
+
+	return (
+		<ul className="dataviews-list-view">
+			{ shownData.map( ( item, index ) => {
+				return (
+					<li key={ getItemId?.( item ) || index }>
+						<div
+							role="button"
+							tabIndex={ 0 }
+							onKeyDown={ onEnter( item ) }
+							className={ classNames(
+								'dataviews-list-view__item',
+								{
+									'dataviews-list-view__item-selected':
+										selection.includes( item.id ),
+								}
+							) }
+							onClick={ () => onSelectionChange( [ item ] ) }
+						>
```

Comment:
Have you tried/considered using [`Button`](https://wordpress.github.io/gutenberg/?path=/docs/components-button--docs) here? Generally better to use existing components rather than reimplementing behaviour.

Reply:
Yeah. Other than styling, my main concern was that this component is going to have buttons within ([this](https://github.com/WordPress/gutenberg/pull/56746#issuecomment-1840500851), but also actions).

Reply:
Hmm... OK. In which case, [we shouldn't be doing this at all](https://accessibleweb.com/question-answer/why-are-nested-interactive-controls-an-accessibility-issue/#:~:text=When%20an%20interactive%20control%20has,information%2C%20or%20interacting%20with%20it.)! Will need to have a think about how best to handle this scenario.

Reply:
> Hopefully it's okay to 'visually nest' those buttons, with either absolute positioning, something like the List view implementation, or some other technique?

Yeah, it's a pretty well established paradigm to visually overlay interactive elements. We just can't nest them in the DOM. For example, this card pattern is routinely used on news sites and in other related content...

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/9e6cd797-b08f-4f84-a6c1-1d40cfe5156c" alt="An article 'card' showing an image, a large title, some text, and a category. The title is underlined with a dark focus ring." width="45%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/daba5525-7937-4a7b-bb4a-249f149720da" alt="An article 'card' showing an image, a large title, some text, and a category. The category is underlined with a dark focus ring." width="45%">
</p>

There are two interactive elements in this card – links for the article title, and the the article category.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/f3fb1d56-8751-4974-9e21-737baae94d86" alt="An article 'card' showing an image, a large title, some text, and a category. The title is underlined, and a mouse pointer is visible in the centre of the card." width="45%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/0ef569dd-8440-4f8a-a189-d1cf78ddbf9a" alt="An article 'card' showing an image, a large title, some text, and a category. The category is underlined, with a mouse pointer visible on top of it." width="45%">
</p>

However, through some clever use of [generated content](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_generated_content), and a bit of `z-index` magic, the entire card becomes clickable, and behaves as if it's all one big link, unless you intentionally hover over the category link, which then becomes the target action.

We don't have to implement it exactly like that, but the general idea is there. One thing to be aware of when doing this is target size – it's bad enough when it's hard to tap on something, but when it's surrounded by a different action context it's even worse.

Reply:
Nice, thanks for sharing the details. It'll be useful when we implement this feature.